### PR TITLE
Fix display name UTF-8 decoding issue

### PR DESCRIPTION
**Pull request by niklasvincent**
# Fix display name UTF-8 decoding issue
## What does this change?

Currently display names containing special characters don't display correctly in the header, which is bad from an internationalisation point of view.

The Identity team is looking at URI encoding the display name before it gets passed to the cookie, which means client side implementations will have to run the display name field through `decodeURIComponent(str)`.

This PR contains the necessary front-end changes to comply with the new encoding.

**Before:**

screen shot 2016-03-29 at 12 58 38

```
window.guardian.config.user
Object {id: "21840015", displayName: "Niklas JÃ´se Lindblad", accountCreatedDate: 1459244427000, emailVerified: true, rawResponse: "WyIyMTg0MDAxNSIsIiIsIk5pa2xhcyBKw7RzZSBMaW5kYmxhZC…zZJOkfDTCcWQOOR7CAtAhQTeije3RGtlpIn3EnVhJDBT0jkuQ"}
```

**After:**

screen shot 2016-03-29 at 12 58 33

```
window.guardian.config.user
Object {id: "21840015", displayName: "Niklas Jôse Lindblad", accountCreatedDate: 1459244427000, emailVerified: true, rawResponse: "WyIyMTg0MDAxNSIsIiIsIk5pa2xhcyBKw7RzZSBMaW5kYmxhZC…zZJOkfDTCcWQOOR7CAtAhQTeije3RGtlpIn3EnVhJDBT0jkuQ"}
```
## What is the value of this and can you measure success?

It provides a nicer user experience by displaying people's names correctly.

User Help has seen a few e-mails on this topic, which should hopefully drop to zero after this fix.
## Does this affect other platforms - Amp, Apps, etc?

Unless Amps uses display name, otherwise no.
## Screenshots

**Before:**

screen shot 2016-03-29 at 12 58 38

**After:**

screen shot 2016-03-29 at 12 58 33
## Request for comment

Would really appreciate feedback on this change.
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: common/app/templates/inlineJS/nonBlocking/getUserData.scala.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ try {
     ((isVeryModern, document, window) => {
         if (isVeryModern) {
             function decodeBase64(str) {
-                return decodeURIComponent(encodeURIComponent(atob(str.replace(/-/g, '+').replace(/_/g, '/').replace(/,/g, '='))));
+                return decodeURIComponent(escape(atob(str.replace(/-/g, '+').replace(/_/g, '/').replace(/,/g, '='))));
```
Comment (jbreckmckye) [inline common/app/templates/inlineJS/nonBlocking/getUserData.scala.js]: Isn't `escape` deprecated?

**Comment by paperboyo** [comment]
:gem:

Finally fixes https://github.com/guardian/frontend/issues/10689. Thanks a lot, @nlindblad!

File: static/test/javascripts/spec/common/identity/Api.spec.js [hunk 2]
```diff
@@ -34,8 +34,8 @@ define([
                 };
                                getCookieStub = sinon.stub();
                                getCookieStub.withArgs('GU_U').returns(
-                                   'WyIyMzEwOTU5IiwiamdvcnJpZUBnbWFpbC5jb20iLCJqYW1lc2dvcnJpZSIsIjUzNCIsMTM4Mjk1MzAzMT' +
-                                   'U5MSwxXQ.MC0CFBsFwIEITO91EGONK4puyO2ZgGQcAhUAqRa7PVDCoAjrbnJNYYvMFec4fAY'
+                                   'WyIyMzEwOTU5IiwiamdvcnJpZUBnbWFpbC5jb20iLCJBbcOpbGllIErDtHNlIiwiNTM0IiwxMzgyOTUzMDMxNTkxLDFd' +
```
Comment (jbreckmckye) [inline static/test/javascripts/spec/common/identity/Api.spec.js]: This won't test the JavaScript you've changed.

In fact, we currently have no easy way to test the inlined scripts, excepting Selenium tests (which we probably don't want to use).

**Comment by johnduffell** [comment]
this is my PR of the day! :grinning: our users will be happy again, thanks

File: static/test/javascripts/spec/common/identity/Api.spec.js (hunk 2 above)
Comment (johnduffell) [inline static/test/javascripts/spec/common/identity/Api.spec.js]: agreed about the selenium tests I wonder what such a test would achieve - if we did break it, we would notice pretty fast!

Comment (nlindblad) [inline static/test/javascripts/spec/common/identity/Api.spec.js]: @jbreckmckye: Yeah, I should add a note on that. I just wanted to make sure that the fallback is doing the right thing as well (which it does already, but without any tests verifying that).

**Comment by ScottPainterGNM** [comment]
:gem:

File: common/app/templates/inlineJS/nonBlocking/getUserData.scala.js (hunk 1 above)
Comment (desbo) [inline common/app/templates/inlineJS/nonBlocking/getUserData.scala.js]: I think we can get rid of it now that we're encoding names in identity.

**Comment by jamespamplin** [comment]
:+1: great stuff! Glad you found a way around using `escape` :)

**Comment by sndrs** [comment]
:100: :star:

**Pull request merged**